src/Application.vala

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1+
+ * SPDX-FileCopyrightText: 2023 elementary, Inc. (https://elementary.io)
+ */
+
+public class Ag.Application : Gtk.Application {
+    public Application () {
+        Object (application_id: "io.elementary.pantheon-agent-polkit");
```

Comment:
```suggestion
        Object (
            application_id: "io.elementary.PolkitAgent",
            flags: GLib.ApplicationFlags.IS_SERVICE,
            register_session: true
        );
```

given that there isn't a app-id before, it's better to define one without dashes (since they aren't valid for DBus paths).

the `register_session` property let us drop the `register_with_session()` dance that is done in the Agent class.

Reply:
Oh hey TIL, thanks for that!